### PR TITLE
Outline Fall/Winter Update!!

### PR DESCRIPTION
**Pull request by himerus**
## Summary by CodeRabbit

- **New Features**
	- Enhanced core link and alert components with updated functionalities and styles.
	- Introduced new documentation and Storybook examples for core components.
	- Updated core button component with new styles and configurations.
- **Documentation**
	- Comprehensive updates across guides and component documentation to reflect new component updates and best practices.
- **Refactor**
	- Streamlined Storybook configuration and documentation structure.
	- Revised PostCSS and ESLint configurations for improved development experience.
- **Chores**
	- Updated Yarn, Changeset, and Git configurations for better version control and package management.
- **Style**
	- Implemented new CSS variables and styles for core components, enhancing UI consistency and customization.

### CODE REVIEW

File: packages/components/outline-core-link/src/outline-core-link.ts [hunk 1]
```diff
@@ -56,6 +61,34 @@ export class OutlineCoreLink extends OutlineElement {
...
+  connectedCallback() {
+    super.connectedCallback();
+    this.adoptedStylesheets = new AdoptedStyleSheets(css`
+      ${componentVars}
+
+      ${globalStyles}
+    `);
+    this.adoptedStylesheets = new AdoptedStyleSheets(globalStyles);
+    this.addController(this.adoptedStylesheets);
+  }
```
[the remark this answers is not in the document]
Comment (himerus) [reply packages/components/outline-core-link/src/outline-core-link.ts]: @coderabbitai I think this is fixed now.

File: packages/components/outline-core-contained-element.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          // Set the style property directly
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          (this.style as any)[propName] = value;
+        } else {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          (this.style as any)[propName] = null;
+        }
+      }
+    });
+  }
```
[the remark this answers is not in the document]
Comment (himerus) [reply packages/components/outline-core-contained-element.ts]: Thanks! This file/component was a sample and has been removed from this PR.

File: packages/components/outline-core-contained-element.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      'pt',
+      'pc',
+    ];
+    const values = value.split(' ').filter(v => v.trim() !== ''); // split by space and filter out any empty strings
+    if (values.length > 4) return false;
+
+    return values.every(
+      v => cssUnits.some(unit => v.endsWith(unit)) || !isNaN(Number(v))
+    );
+  }
```
[the remark this answers is not in the document]
Comment (himerus) [reply packages/components/outline-core-contained-element.ts]: Thanks! This file/component was a sample and has been removed from this PR.

File: packages/documentation/outline-storybook/config/storybook.main.css [hunk 4]
```diff
@@ -281,8 +31,9 @@ hr {
   border-top-width: 1px;
 }
 abbr[title] {
-  -webkit-text-decoration: underline dotted;
-  text-decoration: underline dotted;
+  text-decoration: underline;
+  -webkit-text-decoration: underline dotted currentColor;
+  text-decoration: underline dotted currentColor;
```
[the remark this answers is not in the document]
Comment (himerus) [reply packages/documentation/outline-storybook/config/storybook.main.css]: @coderabbitai can we just let this go for now? We're going to be shifting to using tokens at some point anyway, and this css file is a generated file that was never meant to be committed in the first place.

File: packages/components/outline-core-alert/README.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+| `icon-start` | The icon to display at the start of the alert. |
+
+## CSS Custom Properties
+
+| Property                   | Description                                      |
+|----------------------------|--------------------------------------------------|
+| `--outline-alert--error`   | background: The background color for the error alert. |
+| `--outline-alert--info`    | background: The background color for the info alert. |
+| `--outline-alert--success` | background: The background color for the success alert. |
+| `--outline-alert--warning` | background: The background color for the warning alert. |
```
[the remark this answers is not in the document]
Comment (himerus) [reply packages/components/outline-core-alert/README.md:30]: Appreciated the suggestion, however, this, and other `README.md` files in the Outline repo are auto generated. We will consider this as a future feature.

**Review by shaal — APPROVED** [review]
This is awesome, thank you!

**Comment by glitchgirl** [comment]
TODO:
- Jake get free time 
- Jake write clear directions about what this changes so it feels less scary

**Pull request merged**